iob_soc.py

```diff
@@ -177,88 +177,74 @@ def __init__(self, *args, **kwargs):
             ),
         ]
 
-        # Create system ports and references for them
-        clk_en_rst = port_group(
+        # Create system ports and wires for them
+
+        # This function creates a wire group and adds it to a project global list of wire groups.
+        create_wire_group(
             name="clk_en_rst",
+            wire_prefix="",
+            wires=[],
+            descr="Wire group for global clock, enable, and reset",
+        )
+        # This function creates a port group in the local module's `port_group_list`, and connects it to a wire group from the global list.
+        create_port_group(
             type="slave",
             port_prefix="",
-            wire_prefix="",
             descr="Clock, enable, and reset",
-            ports=[],
+            connect="clk_en_rst",
         )
-        # Extract some individual wires for use in modules below
-        clk = clk_en_rst.wires.clk
-        cke = clk_en_rst.wires.cke
 
-        trap = port(
+        # This function creates a wire named 'trap' and adds it to a project global list of wires.
+        create_wire(
             name="trap",
+            width=1,
+        )
+        # This function creates a port in the local module's `port_group_list`, and assigns a group to it. It also connects the port to a wire from the global list.
+        create_port(
```

Comment:
Creating a port automatically creates a wire since port is subclass of wire.

Don't see need to create a wire and a port.

Reply:
Should we remove the `connect` attribute from the `create_port` and `create_port_group` functions?
If they already create the corresponding port wire automatically, then I think we don't need to specify the wire that connects to it.

Reply:
Removed wires for ports in commit [0cfa061](https://github.com/IObundle/iob-soc/pull/813/commits/0cfa0611c2abe5bf1551a0ea8da33f5b93238487)

Reply:
only groups connect , even if single element groups

Reply:
Should these `create_port` lines be modified?
This is related to this comment: https://github.com/IObundle/iob-soc/pull/813#discussion_r1529434278

Reply:
Port_group should be called simply port, because it is user exposed. The current port should be named port_signal , a more elaborate name but  it is hidden from the user.

Reply:
Should we do the same for wires?
The current `wire_group` becomes just `wire`

Reply:
yes

Reply:
but w cn also use iob_wire and iob_bus, I thnk it is better